### PR TITLE
Don't ship 0 `departure_time` unless specified by user

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Follow-up to https://github.com/ibi-group/datatools-ui/pull/836. Shipping a default 0 departureTime for pattern stop stop times is fine, but for pattern locations this cause unwanted behavior. This PR only ships 0 if the stop time is a pattern stop, otherwise it sends a null.

### CODE REVIEW

File: lib/editor/components/timetable/TimetableEditor.js [hunk 1]
```diff
@@ -216,7 +216,7 @@ export default class TimetableEditor extends Component<Props, State> {
         }
         // }
         // if (stop.timepoint === null || stop.timepoint) {
```
Comment (philip-cline) [inline lib/editor/components/timetable/TimetableEditor.js:218]: I know these are unrelated, but do you think we can remove these comments?

**Review by philip-cline — APPROVED** [review]
Another great 1 liner in the collection

File: lib/editor/components/timetable/TimetableEditor.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/editor/components/timetable/TimetableEditor.js:218]: Git blame says they're from 4 years ago. I would but this file is difficult enough to parse as it is I think removing the historical record of what happened here might hurt more than it helps

**Comment by miles-grant-ibigroup** [comment]
Throwing a blocked on here until we figure out what's going on with these e2e tests

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM

**Pull request merged**